internal/providers/msentraid/msentraid.go

```diff
@@ -83,12 +83,12 @@ func (p Provider) getGroups(token *oauth2.Token) ([]info.Group, error) {
 	cred := azureTokenCredential{token: token}
 	auth, err := msgraphauth.NewAzureIdentityAuthenticationProvider(cred)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to create AzureIdentityAuthenticationProvider: %v", err)
 	}
 
 	adapter, err := msgraphsdk.NewGraphRequestAdapter(auth)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to create GraphRequestAdapter: %v", err)
```

Comment:
Although I  understand the reasoning behind these changes, this messaging is for engineers, not users. Assuming we merge the other PR for redacting the errors, this error message will be logged instead of returned, and it's ok.

But, currently, it does not make much sense for us to return this kind of message to the end user as it is related to implementation details rather than an error that the user will actually understand.

Reply:
Fully agreed, these error messages should be logged, mainly to help us debug the issue, not shown the user in the GDM or terminal.